src/origin_sdk/service/Project.py

```diff
@@ -20,10 +20,90 @@ def get(self, key: str):
         """
         return self.project.get(key)
 
+    def refresh(self):
+        """Refreshes the project data."""
+        self.project = self.session.get_project(self.project_id)
+        return self
+
     def pin(self):
+        """Pins the project, if it isn't already"""
         if not self.project.get("isProjectPinned"):
             self.session.pin_project(self.project_id)
+        return self
 
     def unpin(self):
+        """Unpins the project, if it isn't already"""
         if self.project.get("isProjectPinned"):
             self.session.unpin_project(self.project_id)
+        return self
+
+    def get_or_create_scenario_by_name(
+        self, scenario_name: str, base_scenario_id: Optional[str] = None
+    ):
+        """Either gets a scenario from this project or creates a scenario in
+        this project with the name specified. In the case of create, requires a
+        base_id.
+
+        Arguments:
+            scenario_name: str - The name of the scenario to look for
+            base_id: Optional, str - The ID of the scenario to use as a base
+        """
+        self.refresh()
+
+        scenarios = [
+            scenario
+            for scenario in self.project.get("scenarios")
+            if scenario.get("name").lower() == scenario_name.lower()
+        ]
+
+        scenario = None
+        if len(scenarios) > 0:
+            # We found a scenario with a name match!
+            # Select it to return
+            scenario = scenarios[0]
+        elif base_scenario_id is not None:
+            # No scenario found, create it.
+            scenario = self.session.create_scenario(
+                {
+                    "projectGlobalId": self.project.get("projectGlobalId"),
+                    "baseScenarioGlobalId": base_scenario_id,
+                    "name": scenario_name,
+                }
+            )
+        else:
+            logger.warn(
+                (
+                    "No scenario found in project ",
+                    f"{self.project.get('name')}",
+                    f"' with name '{scenario_name}'",
+                )
+            )
+
+        return scenario
+
+    @staticmethod
+    def get_or_create_project_by_name(
+        session: OriginSession, name: str, pin_project: bool = False
```

Comment:
What Team/Product are these being created under?

Reply:
Also I would have expected two functions again

Reply:
Customer's don't specify teams or products. I will add as a separate optional field.

Re: `get_or_create` it's a useful and I think common(?) pattern but I can read around this more and see if it's confirmation bias.